tests/test_api_facts.py

```diff
@@ -158,6 +158,8 @@ def test_put_facts_with_RBAC_denied(subtests, mocker, api_put, db_create_host, e
 
             assert_response_status(response_status, 403)
 
+            assert db_get_host(host.id).facts != DB_NEW_FACTS
```

Comment:
these would never be equal as on the left side you have all the namespaces whereas on the right side you have data in DB_FACTS_NAMESPACE only

```suggestion
            assert db_get_host(host.id).facts[DB_FACTS_NAMESPACE]["facts"] != DB_NEW_FACTS
```

Reply:
Looks like we weren't using the `api_put` correctly either. Fixed this + the `api_put`. :+1: